src/components/NavBar/SearchBar.tsx

```diff
@@ -0,0 +1,556 @@
+              collection={suggestion as GenieCollection}
+              isHovered={hoveredIndex === index + startingIndex}
+              setHoveredIndex={setHoveredIndex}
+              toggleOpen={toggleOpen}
+              index={index + startingIndex}
+            />
+          ) : (
+            <TokenRow
+              key={suggestion.address}
+              token={suggestion as FungibleToken}
+              isHovered={hoveredIndex === index + startingIndex}
+              setHoveredIndex={setHoveredIndex}
+              toggleOpen={toggleOpen}
+              index={index + startingIndex}
+            />
+          )
+        )}
+      </Column>
+    </Column>
+  )
+}
+
+interface SearchBarDropdownProps {
+  toggleOpen: () => void
+  tokens: FungibleToken[]
+  collections: GenieCollection[]
+  hasInput: boolean
+}
+
+export const SearchBarDropdown = ({ toggleOpen, tokens, collections, hasInput }: SearchBarDropdownProps) => {
+  const [hoveredIndex, setHoveredIndex] = useState<number | undefined>(undefined)
+  const searchHistory = useSearchHistory(
+    (state: { history: (FungibleToken | GenieCollection)[] }) => state.history
+  ).slice(0, 2)
+  const isNFTPage = location.hash.includes('/nft')
+  const isTokenPage = location.hash.includes('/token')
+
+  const tokenSearchResults =
+    tokens.length > 0 ? (
+      <SearchBarDropdownSection
+        hoveredIndex={hoveredIndex}
+        startingIndex={isNFTPage ? collections.length : 0}
+        setHoveredIndex={setHoveredIndex}
+        toggleOpen={toggleOpen}
+        suggestions={tokens}
+        header={'Tokens'}
+      />
+    ) : (
+      <Box className={styles.notFoundContainer}>No tokens found.</Box>
+    )
+
+  const collectionSearchResults =
+    collections.length > 0 ? (
+      <SearchBarDropdownSection
+        hoveredIndex={hoveredIndex}
+        startingIndex={isNFTPage ? 0 : tokens.length}
+        setHoveredIndex={setHoveredIndex}
+        toggleOpen={toggleOpen}
+        suggestions={collections}
+        header={'NFT Collections'}
+      />
+    ) : null
+
+  // TODO Trending NFT Results implmented here
+  const trendingCollections = [] as TrendingCollection[]
+
+  const { data: trendingTokenResults } = useQuery([], () => fetchTrendingTokens(4), {
+    refetchOnWindowFocus: false,
+    refetchOnMount: false,
+    refetchOnReconnect: false,
+  })
+
+  const trendingTokens = useMemo(() => {
+    // TODO reimplement this logic with NFT search
+    // return trendingTokenResults?.slice(0, isTokenPage ? 3 : 2)
+    return trendingTokenResults?.slice(0, 4)
+  }, [trendingTokenResults])
+
+  const totalSuggestions = hasInput
+    ? tokens.length + collections.length
+    : Math.min(searchHistory.length, 2) +
+      (isNFTPage || !isTokenPage ? trendingCollections?.length ?? 0 : 0) +
+      (isTokenPage || !isNFTPage ? trendingTokens?.length ?? 0 : 0)
+
+  // Close the modal on escape
+  useEffect(() => {
+    const keyDownHandler = (event: KeyboardEvent) => {
```

Comment:
would love to move the keyboard hotkey stuff out since you have a few in here. perhaps a `useHotKeyHandler` hook?

Reply:
We need some the keyboard logic to be localized to the suggestion row components, especially for tracking hovered/focused state, which is why we cant abstract to a generalized hook. You can read more on our decision when this was developed in this pr: https://github.com/Uniswap/interface-nft/pull/26

Reply:
optional